src/revlanguage/workspace/RbRegister_Dist.cpp

```diff
@@ -607,6 +608,25 @@ void RevLanguage::Workspace::initializeDistGlobalWorkspace(void)
         AddDistribution< ModelVector<Integer>       >( new Dist_IID<Integer>()      );
         AddDistribution< ModelVector<Probability>   >( new Dist_IID<Probability>()  );
 
+        // Inverse distribution
+        AddDistribution< Integer                    >( new Dist_Inverse<Integer>() );
+        AddDistribution< Natural                    >( new Dist_Inverse<Natural>() );
+        AddDistribution< Probability                >( new Dist_Inverse<Probability>() );
+        AddDistribution< Real                       >( new Dist_Inverse<Real>() );
+        AddDistribution< RealPos                    >( new Dist_Inverse<RealPos>() );
+        AddDistribution< Simplex                    >( new Dist_Inverse<Simplex>() );
+        AddDistribution< AbstractHomologousDiscreteCharacterData  >( new Dist_Inverse< AbstractHomologousDiscreteCharacterData >());
+        AddDistribution< ModelVector<Integer>       >( new Dist_Inverse< ModelVector<Integer> >());
+        AddDistribution< ModelVector<Natural>       >( new Dist_Inverse< ModelVector<Natural> >());
+        AddDistribution< ModelVector<Probability>   >( new Dist_Inverse< ModelVector<Probability> >() );
+        AddDistribution< ModelVector<Real>          >( new Dist_Inverse< ModelVector<Real> >());
+        AddDistribution< ModelVector<RealPos>       >( new Dist_Inverse< ModelVector<RealPos> >());
+        AddDistribution< ModelVector<TimeTree>      >( new Dist_Inverse< ModelVector<TimeTree> >());
+        AddDistribution< ModelVector< ModelVector<TimeTree> >                  >( new Dist_Inverse< ModelVector<ModelVector<TimeTree> > >());
+        AddDistribution< ModelVector<BranchLengthTree>                         >( new Dist_Inverse< ModelVector<BranchLengthTree> >());
+        AddDistribution< ModelVector<AbstractHomologousDiscreteCharacterData>  >( new Dist_Inverse< ModelVector<AbstractHomologousDiscreteCharacterData> >());
+//        AddDistribution< AbstractHomologousDiscreteCharacterData >( new Dist_phyloCTMCClado() );
+
```

Comment:
Is there an obvious workaround for this?  Given that `AbstractHomologousDiscreteCharacterData` didn't work "out of the box", it's possible that further work would be needed to ensure that these other cases are handled correctly.  If we are going to support these, I would be most comfortable creating a test case for each supported type.  My inclination would be to hold off from doing this work until we can see a clear use case for these other types.

Reply:
I think the way RevBayes is currently constructed, a lot of things require that all possible types are registered, even if we are doing the "same" thing for each one.  It is indeed annoying, but I think there are a lot of cases where this occurs, and we generally don't really list all possible types in each case.  So maybe this could wait for a use case to add more types.

Reply:
Maybe I'm just being too paranoid about composability... so many of our issues come from users trying to combine existing features in unforeseen ways (which we generally encourage) and finding out that it doesn't work. Waiting for a use case sounds to me like waiting for this to happen.

Do I understand correctly that the problem with `AbstractHomologousDiscreteCharacterData` is that it is an abstract class, and that's why it requires special handling? If so, would it be a reasonable middle ground to support all types that are not implemented as abstract classes?

Reply:
Well, take a look at the rest of the file.  Dist_mixture<T> is only defined for 9 types.  It is defined for ModelVector<Real>, but not ModelVector<ModelVector<Real>>.  

Dist_IID<T> is defined for 5 types.

Dist_EmpiricalSample< > is defined for 8 types.

I'm not so much saying that this is fine, as that is not a NEW problem.

The right solution might be to do generic type handling at runtime, instead of trying to generate all possible types using templates at compile time.  :man_shrugging:

Reply:
Yes, now that you've proposed the modification elsewhere to handle all `Cloneable` types rather than making a special case just for `AbstractHo...`, I feel more confident that the function should in principle support most other types more broadly.  

If I've understood, though, there's no way to say `AddDistribution< Cloneable >` other than to list all cloneable types one by one in this file.  If that's desirable, is there a list I can draw on for all types that it might be useful to support?  (In principle, I suppose it would make sense already to handle non-discrete character data, for example.)

Reply:
Given that its hard to come up with a second example where we want to use dnInverse, perhaps its premature to add all these other types...  If this was a general solution to ascertainment bias, then I would lean more towards implementing this for all those other types.

Reply:
Happy to move ahead with this – if the real-world use cases for `dnInverse` are rather limited (at least for now), there is probably no point in aiming for full generality right from the start.